### ISSUE TITLE
After app has sat idle switching to new dashboard pages causes the app to hang for 30-45s (iPadOS)

### ISSUE THREAD

**Issue by KidA001** [home-assistant/iOS#2812]
**iOS device model, version and app version**
iPad (6th Generation)
Model MR7F2LL/A
iPadOS 17.5.1

Homeassistant App version:  2024.05

**Home Assistant Core Version**
`2024.5.5`

**Describe the bug**
I run my iPad in Kiosk mode, it always has the HA app open. Up until this point I have never had issues. When the app has been sitting for a while (30m to an 1-hr) and I tap a new dashboard page it hangs and doesn’t fully render the page. I either have to wait 30-45s or force quit and re-open. When I force quit and re-open everything works as expected and I can navigate through pages, but the same issue persists after it has sat for a while.

It happens regardless of what dashboard I open. Simple page vs more complex page which graphs, they all will hang.

I’ve taken a video of what happens which you can see here: [PXL_20240530_000001942.mp4 - Google Drive](https://drive.google.com/file/d/1E92YtSKr1hNlWAaYN9jWjcgVissHaEgM/view?usp=sharing)

**To Reproduce**
I can repro this after my app has sat open and untouched for 30min to 1-hour. I have factory reset my iPad and re-installed the HomeAssistant app and this issue still persists.

**Expected behavior**
It doesn't hang when I go to new dashboard pages. This was not an issue for me until this most recent update.

**Screenshots**
I’ve taken a video of what happens which you can see here: [PXL_20240530_000001942.mp4 - Google Drive](https://drive.google.com/file/d/1E92YtSKr1hNlWAaYN9jWjcgVissHaEgM/view?usp=sharing)

**Comment by KidA001** [comment]
This issue still persists.

**Comment by KidA001** [comment]
@bgoncal any other info I can provide to help diagnose this?

**Comment by KidA001** [comment]
Issue still persists and has made my iPad dashboard unusable when switching dashboard pages. by the time it loads i've already opened my phone to do whatever it is needed to do.

**Comment by bgoncal** [comment]
Hi @KidA001, before moving forward can you try using an older version of the App? Use testflight to navigation to an older app version and see if it behaves the same, otherwise then we have to consider a frontend/core update may have caused this.

**Comment by KidA001** [comment]
@bgoncal  I'm familiar with testflight, do you need to assign me access to get different versions of the app or can you point me to where I'll find details for this? Thank!

**Comment by bgoncal** [comment]
Take a look here:

![image](https://github.com/home-assistant/iOS/assets/5808343/aad70481-94d4-437c-b643-5c89b4310209)

**Comment by KidA001** [comment]
@bgoncal  I need to be invited to test HomeAssistant on Testflight though. I have testflight installed but you need to add my ID I believe. I'm `KidA001(at)gmail.com`

**Comment by KidA001** [comment]
Yes I'm familiar with that, I wasn't sure how to be invited to test the HA app in testflight. I see you need to go to https://testflight.apple.com/join/1AlPbnLZ

I just installed 2024.4.1 and will report back, thank you.

**Comment by KidA001** [comment]
@bgoncal I can confirm that the issue does not persist in `2024.4.1`, when I switch between dashboards after the app has sat for an hour it's not hanging like it does is `2024.5.x`

**Comment by bgoncal** [comment]
Please also try 2024.6

**Comment by KidA001** [comment]
@bgoncal I tried 2024.6 overnight and the issue is present. Reverting to 2024.4 for now - Any thoughts on what it might be? Any other details I can provide to help debug?

**Comment by bgoncal** [comment]
I need to check what's new since 2024.4 and evaluate what might have caused that.

You don't have a crash at any moment right?

**Comment by KidA001** [comment]
No crash, just the lagging when switching views on the dashboard as shown in the video. It can be up to 30-45 seconds before the view becomes available to interact with.

**Comment by bgoncal** [comment]
Can you tell me which exaclt is the last version that works well for you? You said 2024.4.1, can you tell the build number? (someting like 2024.XXX after the version)

**Comment by KidA001** [comment]
the last working build seems to be `2024.677`

**Comment by bgoncal** [comment]
While I investigate what may have caused the issue, can you create a fresh dashboard and try to reproduce the issue there using the minimum cards possible? Just to we evaluate the relation to cards/webview performance

**Comment by bgoncal** [comment]
Quickly went over our thread again and I dont think you have tried to clean front end cache yet right? Companion App settings >> debugging >> reset frontend cache

**Comment by KidA001** [comment]
> Quickly went over our thread again and I dont think you have tried to clean front end cache yet right? Companion App settings >> debugging >> reset frontend cache

I have factory reset the iPad and reinstalled iOS and HA before posting this GH Issue to ensure it wasn't the iPad. The issue still persisted. I'll reset the frontend cache just to be sure.

Sure thing, I'll create a new dashboard and use the latest build and see if it has the same behavior. I'm also a dev, if there's any more advanced debugging I can help with let me know.

**Comment by KidA001** [comment]
I am on `2024.7` build `2024.700` and I also reset the frontend cache. It seems to be working but I'm going to wait a couple hours to verify.

Is resetting the frontend cache a serverside/host cache reset or is it local app cache?

**Comment by KidA001** [comment]
Just to confirm I've installed 2024.05 which previously had the issue I've been describing here, and it seems to be working fine. I'm going to let it run overnight to confirm before I close. I'm curious again though if resetting the frontend cache is a local or serverside reset and before I posted this I had completely erased my ipad and reinstalled everything to confirm it wasn't an issue like that.

I'll report back.

**Comment by bgoncal** [comment]
It's just a local web data store reset 

<img width="726" alt="Screenshot 2024-07-02 at 11 08 49" src="https://github.com/home-assistant/iOS/assets/5808343/7dd161c2-4e9d-429d-a6f4-a0989c9e2aa8">

**Comment by bgoncal** [comment]
I couldn't reproduce on my side so... no questions for now, let me know in case it happens again.

**Comment by KidA001** [comment]
@bgoncal can we re-open this issue? The same problem is happening, clearing the cache will temporarily solve it (not fully) but the more days the app sits without clearing the cache the more lag happens after switching views

**Comment by villasenor** [comment]
I'm having the same issue @KidA001 saw on an iPhone running iOS 17.6.1, Public release version 2024.9.1 of the app. Reset frontend cache fixes the issue for literally a minute, but then opening the hamburger menu starts getting slow again, and the app slows down. I remember this being an issue on slow cellular connections a while back, but I never remember things being this slow on a local connection with a great Wi-Fi connection. I'm also a dev - let me know how I can help @bgoncal, and thanks for all your hard work on the iOS Companion App!

**Comment by bgoncal** [comment]
@villasenor which Core version are you using? Can you share a screen recording ?

**Comment by tenallero** [comment]
Hi,

I have several devices
iPad Pro (2nd gene) with iOS 16.7.10
iPad (5th gen) with iOS 16.7.10
iPhone 11 with iOS 17
iPhone 12 with iOS 18

My Home Assistant Core release is 2024.10.1
I use Nabucasa Cloud for external URL

Since june/july (approximately) HomeAssistant companion has become much slower. And on older devices (iPad & iPadPro) it crashes after a few seconds. I don't know if this is due to companion or iOS upgrades.

Regarding older devices (iPads) I have tried everything. Delete the app with data, reset the device to factory defaults. Run with TestFlight to the oldest release (2024.7)
The results are always the same. Once the app has connected to HA ... the app freezes to crash. I can't even use the app settings options, obtain logs, etc.
